### ISSUE TITLE
The usefulness of _output.set()

### ISSUE THREAD

**Issue by BoPeng** [vatlab/SoS#1138]
`group_with` sets variables that are pre-determined from the input statement and existing variables. However, the following usage can potentially be very useful: 

```
[10]
input: for_each=dict(i=range(5))
output: f'res_{i}.txt'
import random
seed = random.seed()

_output.set(seed=seed)
```
That is to say, to attach computation results to the result, which, is used properly, could be a better way than using the `shared` section option.

So, although we can disable the change of content of `_output`, we might want to allow the setting of attributes to `_output` and its children.

**Comment by gaow** [comment]
Yes I am in favor of the change, if there are no other complications involved. In fact I was under the impression it already worked (#1130).

**Comment by BoPeng** [comment]
No, it does not work because of #1137 .

**Comment by BoPeng** [comment]
Yes things need to be pickled around and can be slow if the variables are large, but in a majority of cases output does not need to be returned.
